xla/service/gpu/runtime/dynamic_slice_thunk.cc

```diff
@@ -47,14 +47,28 @@ limitations under the License.
 namespace xla {
 namespace gpu {
 
+namespace {
+
+std::unique_ptr<Literal>& Indvar(DynamicSliceThunk* thunk) {
+  static thread_local absl::flat_hash_map<DynamicSliceThunk*,
+                                          std::unique_ptr<Literal>>
```

Comment:
Why can't we just make `DynamicSliceThunk` hold a pointer to a literal, as opposed to having a `static thread_local` map?

This will outlive the existence of a thunk, and could potentially lead to some kind of UAF if a `DynamicSliceThunk` gets freed and another gets allocated in the same place.

Am I missing something?

Reply:
I tried that, but the same object in the memory is used by multiple threads. I think these threads correspond to the replicas but I am not too sure. Meaning, when the first replica increments the value, all the other replicas also see the new value, and this leads to inconsistency. This is similar to the work here: https://github.com/shraiysh/xla/blob/e27a06fe71dbecc79c2ee636abd4bf23755bd7d6/xla/service/gpu/runtime/while_thunk.cc#L48. Is there a better solution to this problem?